File: JavaPractice/src/ReverseLinkedListValueInteger.java

public class ReverseLinkedListValueInteger {

	static int sum=0;
	static String sb = "";
	static Node head;
	static class Node
	{
		int data;
		Node next;
		
		Node(int d)
		{
			data = d;
			next = null;
		}
	}
	static void reverseValuetoString(Node node) {
		if(node == null)
			return;
		reverseValuetoString(node.next);
		appendString(node.data);
	}
	
	static String appendString(int data) {
		return sb+=data;
		
	}
	static void reverseValues(Node node) {
		if(node == null)
			return;
		reverseValues(node.next);
		computeValues(node.data);
	}
	
	static void computeValues(int val) {
		sum = (sum*100) + val;
	}
	Node ReverseList(Node node)
	{
		Node prev = null;
		Node current = head;
		Node next = null;
		while(current != null)
		{
			next = current.next;
			current.next = prev;
			prev = current;
			current = next;
		}
		node = prev;
		return node;
	}
    void printlist(Node node)
	{
		while(node != null)
		{
			System.out.println(node.data + " ");
			node = node.next;
		}
		
	}
    public Node reverse(Node node)
    {
        if(node == null || node.next == null)
            return node;
        
        Node current = node.next;
        node.next = null;
        Node head = reverse(current);
        current.next = node; 
        
        return head; 
    }
	public static void main(String[] args) {
		// TODO Auto-generated method stub
		ReverseLinkedListValueInteger rl = new ReverseLinkedListValueInteger();
		head = new Node(53);
		head.next = new Node(41);
		head.next.next = new Node(23);
		head.next.next.next = new Node(21);
		reverseValuetoString(head);
		System.out.println(sb);
		int n = Integer.parseInt(sb);
		System.out.println(n);
		/*
		System.out.println("Original Linked list is :");
        rl.printlist(head);
        head = rl.reverse(head);
        System.out.println("");
        System.out.println("Reversed linked list : ");
        rl.printlist(head);
        */
	}

}

File: PythonPractice/src/ReverseLinkedListValueInteger.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class ReverseLinkedListValueInteger:
    def __init__(self):
        self.sum = 0
        self.sb = ""
        self.head = None

    def reverse_valueto_string(self, node):
        if node is None:
            return
        self.reverse_valueto_string(node.next)
        self.append_string(node.data)

    def append_string(self, data):
        self.sb += str(data)

    def reverse_values(self, node):
        if node is None:
            return
        self.reverse_values(node.next)
        self.compute_values(node.data)

    def compute_values(self, val):
        self.sum = (self.sum * 100) + val

    def reverse_list(self, node):
        prev = None
        current = node
        while current is not None:
            next_node = current.next
            current.next = prev
            prev = current
            current = next_node
        return prev

    def print_list(self, node):
        while node is not None:
            print(node.data, end=" ")
            node = node.next
        print()

    def reverse(self, node):
        if node is None or node.next is None:
            return node

        current = node.next
        node.next = None
        head = self.reverse(current)
        current.next = node

        return head

def main():
    rl = ReverseLinkedListValueInteger()

    # Initialize linked list
    rl.head = Node(53)
    rl.head.next = Node(41)
    rl.head.next.next = Node(23)
    rl.head.next.next.next = Node(21)

    # Reverse values to string
    rl.reverse_valueto_string(rl.head)
    print(rl.sb)  # Output the reversed value as a string

    # Convert string to integer and print
    n = int(rl.sb)
    print(n)  # Output the integer value

    # Print original linked list
    print("Original Linked list is :")
    rl.print_list(rl.head)

    # Reverse the linked list
    rl.head = rl.reverse(rl.head)

    # Print reversed linked list
    print("")
    print("Reversed linked list : ")
    rl.print_list(rl.head)

if __name__ == "__main__":
    main()
